test/parallel/test-http2-client-settings-errors.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const {
+  constants,
+  Http2Session,
+  nghttp2ErrorString
+} = process.binding('http2');
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const http2 = require('http2');
+
+// tests error handling within requestOnConnect
+// - NGHTTP2_ERR_NOMEM (should emit session error)
+// - every other NGHTTP2 error from binding (should emit session error)
+
+const specificTestKeys = [
+  'NGHTTP2_ERR_NOMEM'
+];
+
+const specificTests = [
+  {
+    ngError: constants.NGHTTP2_ERR_NOMEM,
+    error: {
+      code: 'ERR_OUTOFMEMORY',
+      type: Error,
+      message: 'Out of memory'
+    }
+  }
+];
+
+const genericTests = Object.getOwnPropertyNames(constants)
+  .filter((key) => (
+    key.indexOf('NGHTTP2_ERR') === 0 && specificTestKeys.indexOf(key) < 0
+  ))
+  .map((key) => ({
+    ngError: constants[key],
+    error: {
+      code: 'ERR_HTTP2_ERROR',
+      type: Error,
+      message: nghttp2ErrorString(constants[key])
+    }
+  }));
+
+const tests = specificTests.concat(genericTests);
+
+runTest(tests.shift());
+
+function runTest(test) {
+  const server = http2.createServer(common.mustNotCall());
+  server.listen(0, common.mustCall(() => {
+    // mock submitSettings because we only care about testing error handling
+    Http2Session.prototype.submitSettings = () => test.ngError;
+
+    const errorMustCall = common.expectsError(test.error, 2);
+    const errorMustNotCall = common.mustNotCall(
+      `${test.error.code} should emit on ${test.type}`
```

Comment:
`test.type` does not appear to be defined in this test. I think it should be replaced with a plain string that says `session`.

Reply:
Great catch @apapirovski 
I'd copied this template from other tests which checks for errors which had types. I've replace `${test.type}` with plain string `session`